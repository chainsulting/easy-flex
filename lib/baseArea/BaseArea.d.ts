import React, { AreaHTMLAttributes } from 'react';
import { Color } from '../types';
import { BorderProps } from '../utils/border';
import { ColorProps } from '../utils/color';
import { DistanceProps } from '../utils/distance';
import { FlexItemProps } from '../utils/flexItem';
import { FontProps } from '../utils/font';
import { SizeProps } from '../utils/size';
export interface BaseAreaProps extends Omit<AreaHTMLAttributes<HTMLAreaElement>, 'color'>, BorderProps, ColorProps, DistanceProps, FlexItemProps, FontProps, SizeProps {
    placeholderColor?: Color;
}
export declare const BaseArea: React.ForwardRefExoticComponent<BaseAreaProps & React.RefAttributes<HTMLAreaElement>>;
