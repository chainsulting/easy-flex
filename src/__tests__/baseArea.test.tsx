import { render } from '@testing-library/react';
import 'jest-styled-components';
import React from 'react';
import { BaseArea } from '../baseArea/BaseArea';

describe('<BaseArea />', () => {
	test('should have the basic css settings', async () => {
		const result = render(<BaseArea />);
		expect(result.baseElement).toMatchSnapshot();
	});
});
