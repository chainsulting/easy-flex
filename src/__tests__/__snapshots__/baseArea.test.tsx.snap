// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseArea /> should have the basic css settings 1`] = `
.c0 {
  box-sizing: border-box;
  border: none;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
}

.c0 {
  box-sizing: border-box;
  border: none;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
}

.c0 {
  box-sizing: border-box;
  border: none;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
}

.c0 {
  box-sizing: border-box;
  border: none;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  padding-top: 0px;
}

<body>
  <div>
    <area
      class="c0"
      data-margin-bottom="0px"
      data-margin-left="0px"
      data-margin-right="0px"
      data-margin-top="0px"
      data-padding-bottom="0px"
      data-padding-left="0px"
      data-padding-right="0px"
      data-padding-top="0px"
    />
  </div>
</body>
`;
